src/Framework.Configurator/configurator-ui/src/app/rights/components/principals/components/grant-rights-dialog/grant-rights-dialog.service.ts

```diff
@@ -162,27 +163,27 @@ export class GrantRightsDialogService {
   }
 
   private getResult(): IGrantedRight[] {
```

Comment:
`getResult()` name is too generic. does `mapPermissionsToGrantedRights` sound good?

also, it can be splitted into
`private mapToGrantedRight(permission: IPermissionUiDto): IGrantedRight {}`
```
private mapContexts(contexts: IContextDto[]): { Id: string; Entities: string[] }[] {
    return contexts.map(({Id, Entities}) => ({Id, Entities: Entities.map(({Id}) => Id)}));
  }
```

Reply:
also, `IsVirtual` filter may be applied after the mapping process

Reply:
there is a one small responsibility here - map (yes and one more is filter) - but split it for a few smaller including each other functions I think is not neccessary


> also, IsVirtual filter may be applied after the mapping process

it's already after mapping, but it can be done before